shell/src/main/java/alluxio/cli/validation/StorageSpaceValidationTask.java

```diff
@@ -0,0 +1,166 @@
+      String rawDirQuota = Configuration.get(tierDirCapacityConf);
+      if (rawDirQuota.length() <= 0) {
+        System.err.format("Tier %d: Quota cannot be empty.%n", level);
+        return false;
+      }
+
+      String[] dirQuotas = rawDirQuota.split(",");
+
+      try {
+        Map<String, MountedStorage> storageMap = new HashMap<>();
+        File file = new File(dirPaths[0]);
+        if (dirPaths.length == 1 && alias.equals("MEM") && !file.exists()) {
+          // skip checking if RAM disk is not mounted
+          System.out.format("RAM disk is not mounted at %s, skip validation.%n", dirPaths[0]);
+          continue;
+        }
+
+        for (int i = 0; i < dirPaths.length; i++) {
+          int index = i >= dirQuotas.length ? dirQuotas.length - 1 : i;
+          long quota = FormatUtils.parseSpaceSize(dirQuotas[index]);
+          success &= addDirectoryInfo(dirPaths[i], quota, storageMap);
+        }
+
+        for (Map.Entry<String, MountedStorage> storageEntry : storageMap.entrySet()) {
+          MountedStorage storage = storageEntry.getValue();
+          long capacity = storage.getCapacitySize();
+          long used = storage.getWorkspaceSize();
+          long available = storage.getAvailableSize();
+          if (capacity > used + available) {
+            System.err.format(
+                "Tier %d: Not enough space on %s. %n"
+                    + "Required capacity: %s%n"
+                    + "Used in working directory: %s%n"
+                    + "Available: %s%n",
+                level, storageEntry.getKey(),
+                FormatUtils.getSizeFromBytes(capacity),
+                FormatUtils.getSizeFromBytes(used),
+                FormatUtils.getSizeFromBytes(available));
+            success = false;
+          }
+        }
+      } catch (IOException e) {
+        System.err.format("Tier %d: Unable to validate available space - %s.%n",
+            level, e.getMessage());
+        success = false;
+      }
+    }
+
+    return success;
+  }
+
+  private boolean addDirectoryInfo(String path, long quota, Map<String, MountedStorage> storageMap)
+      throws IOException {
+    File file = new File(path);
+    if (!file.exists()) {
+      System.err.format("Path %s does not exist.%n", path);
+      return false;
+    }
+
+    if (!file.isDirectory()) {
+      System.err.format("Path %s is not a valid directory.%n", path);
+      return false;
+    }
+
+    long directorySize = FileUtils.sizeOfDirectory(file);
+
+    FileStore store = Files.getFileStore(Paths.get(path));
+
+    MountedStorage storage = storageMap.get(store.name());
+    if (storage == null) {
+      storage = new MountedStorage(store);
+      storageMap.put(store.name(), storage);
+    }
+
+    storage.addCapacitySize(quota);
+    storage.addWorkspaceSize(directorySize);
+    return true;
+  }
+
+  private final class MountedStorage {
+    private long mCapacitySize;
+    private long mWorkspaceSize;
```

Comment:
Is this supposed to be the existing used bytes in the tiered storage?

Are these supposed to be in bytes?